playbooks/tasks/portal-mongo-replicaset-check-majority.yml

```diff
@@ -0,0 +1,89 @@
+---
+
+# Check there is still MongoDB majority if we turn off hosts in the current batch
+
+- name: Check mongo container is running
+  community.docker.docker_container_info:
+    name: mongo
+  register: mongo_docker_container_result
+
+# If the mongo container is already stopped we can skip MongoDB replicaset
+# majority check, it doesn't matter that we stop the rest of services.
+- block:
+    # Set mongo db config from .env file (without using LastPass/secrets storage)
+
+    - name: Get mongo user on host_item from .env file
+      ansible.builtin.shell: ". {{ webportal_dir }}/.env && echo $SKYNET_DB_USER"
+      register: skynet_db_user_result
+
+    - name: Get mongo pass on host_item from .env file
+      ansible.builtin.shell: ". {{ webportal_dir }}/.env && echo $SKYNET_DB_PASS"
+      register: skynet_db_pass_result
+
+    - name: Set mongo db config
+      set_fact:
+        mongodb_config:
+          skynet_db_user: "{{ skynet_db_user_result.stdout }}"
+          skynet_db_pass: "{{ skynet_db_pass_result.stdout }}"
+
+    - name: Include checking mongo replicaset majority
+      include_tasks: tasks/portal-role-task-mongo-shell-eval.yml
+      vars:
+        mongodb_shell:
+          eval: |
+            votingMembers = rs.config().members
+              .filter(m => m.priority > 0)
+              .map(m => m.host)
+            onlineVotingMembers = rs.status().members
+              .filter(m => m.stateStr == 'PRIMARY' || m.stateStr == 'SECONDARY')
+              .map(m => m.name)
+              .filter(m => votingMembers.includes(m))
+            batchMembers = [{% for h in ansible_play_batch -%}
+              '{{ hostvars[h].ansible_host + ':' + (hostvars[h].subcrap_mongo_port | default(default_mongo_port) | string) }}'{{ loop.last | ternary('', ', ') }}
+              {%- endfor -%}]
+            onlineVotingMembersAfter = onlineVotingMembers.filter(m => !batchMembers.includes(m))
+            result = {
+              "votingMembers": votingMembers,
+              "onlineVotingMembers": onlineVotingMembers,
+              "batchMembers": batchMembers,
+              "onlineVotingMembersAfter": onlineVotingMembersAfter,
+              "replicasetMajorityOk": onlineVotingMembersAfter.length > votingMembers.length/2
+            }
+          until: True
+          retries: 0
+
+    - name: Set mongo replicaset result
+      set_fact:
+        mongo_replicaset_result: "{{ mongo_shell_result.transformed_output | from_json }}"
+
+    - name: Stop the playbook if we don't have majority (> 50%) MongoDB replicaset voting members online
```

Comment:
The first thing this playbook should do is check whether the current node is a voting member or not. If it's not a voting member then we don't need to check anything, we can just proceed.

Reply:
Updated fail condition not to fail for non-voting member.